x-pack/plugins/log_explorer/public/customizations/log_explorer_profile.tsx

```diff
@@ -115,6 +116,15 @@ export const createLogExplorerProfileCustomizations =
           viewSurroundingDocument: { disabled: true },
         },
       },
+      Content: (props) => {
+        const KibanaContextProviderForPlugin = useKibanaContextForPluginProvider(core, plugins);
+
+        return (
+          <KibanaContextProviderForPlugin>
+            <LazyCustomFlyoutContent {...props} />
+          </KibanaContextProviderForPlugin>
+        );
+      },
```

Comment:
```suggestion
      Content: (props) => {
        const KibanaContextProviderForPlugin = useKibanaContextForPluginProvider(core, plugins);
        const internalState = useObservable(
          stateContainer.internalState.state$,
          stateContainer.internalState.get()
        );

        return (
          <KibanaContextProviderForPlugin>
            <LazyCustomFlyoutContent {...props} dataView={internalState.dataView} />
          </KibanaContextProviderForPlugin>
        );
      },
```
Instead of adding `dataView` to `FlyoutContentProps`, could we instead retrieve it from the state container directly and pass it to your custom component? I'd prefer to keep a single source of truth for customizations to access the shared state if possible.

On a side note, requiring `useObservable` to access our state in customization components feels a bit inconvenient IMO. Internally we use state selector functions instead, and we could likely give customizations access to the same selectors by passing them to the `CustomizationCallback`. Just curious if it would be useful for your team to have access to these selectors too?

Reply:
Hey Davis, our team is working on extracting the persistent system on top of the state container, so that each consumer of the log explorer won't depend on the URL sync. I think we'll consume the internal state reacting to the observable changes anyway, I can't say if we'll need those selectors.

Regarding your suggestion I see your point, I'll apply your suggestion and revert the changes on the customization contract.

Reply:
Changes applied in https://github.com/elastic/kibana/pull/169832/commits/c0cf9dc1885b8788633dc13b54edb054836d45d4